### PR TITLE
[i2c,dv] Capture progress of transfer into i2c_item.state field

### PR DESCRIPTION
**Pull request by hcallahan-lowrisc**
> For reviewing purposes, ignore the first commit. It is part of #23911, and will be rebased away when that PR is merged.

This change adds more granular transaction tracking into the i2c_item type. A new `.state` field is added, which is updated by the i2c_monitor as a transfer progresses.
This more-granular measurement allows:
1) For the agent's reactive sequences to monitor in-progress transfers directly, and to drive at the appropriate points.
2) For the I2C predictor/ref_model to make more accurate predictions about inputs to the system mid-transfer. Currently we make no use of this, but it will be built upon to start modelling DUT features more precisely.

### CODE REVIEW

File: hw/dv/sv/i2c_agent/i2c_item.sv
Comment (marnovandermaas) [inline hw/dv/sv/i2c_agent/i2c_item.sv:83]: Nit: Alignment looks a bit weird to me here.

File: hw/dv/sv/i2c_agent/i2c_agent_pkg.sv
Comment (marnovandermaas) [inline hw/dv/sv/i2c_agent/i2c_agent_pkg.sv:39]: It might be good to put a short comment on what each state means. For example, what is the difference between `StAddrByte` and `StAddrByteRcvd`?

Comment (marnovandermaas) [reply hw/dv/sv/i2c_agent/i2c_agent_pkg.sv:39]: After reading the next step, maybe this is not necessary. You could just reference the monitor logic.

**Review by marnovandermaas — APPROVED** [review]
Looks like a good change to me from a code review. Just a few small comments but none are blocking. I had a look at the CI failures and they don't look related but it would be good for another set of eyes to confirm that before merging.

File: hw/dv/sv/i2c_agent/i2c_agent_pkg.sv
Comment (hcallahan-lowrisc) [reply hw/dv/sv/i2c_agent/i2c_agent_pkg.sv:39]: Thanks Marno, I think the best thing to do here might be a diagram in the documentation with an annotated i2c waveform showing how each enum state lines up with phases of a transfer. I'm going to file as a future improvement and come back to it.

File: hw/dv/sv/i2c_agent/i2c_item.sv
Comment (hcallahan-lowrisc) [reply hw/dv/sv/i2c_agent/i2c_item.sv:83]: Ah yes, it aligns with some other items just slightly further down the page :)

**Comment by hcallahan-lowrisc** [comment]
The CI failures are unrelated to this PR. Hence I'm going to merge it now.

**Pull request merged**
